Dockerfile

```diff
@@ -40,21 +40,20 @@ ENV POSTGRES_JDBC_VER=42.2.23
 COPY build.gradle settings.gradle gradlew /gradle/
 COPY gradle /gradle/gradle
 ENV GRADLE_JARS_DIR=gradle_jars
-RUN /gradle/gradlew -p /gradle build
-RUN cp -r /gradle/${GRADLE_JARS_DIR}/* /opt/bitnami/spark/jars/
+RUN /gradle/gradlew -p /gradle build && \
+    cp -r /gradle/${GRADLE_JARS_DIR}/* /opt/bitnami/spark/jars/ && \
+    rm -rf /gradle
 
 # make an empty yarn conf dir to prevent spark from complaining
 RUN mkdir -p /opt/yarn/conf && chown -R spark_user:spark /opt/yarn
 ENV YARN_CONF_DIR=/opt/yarn/conf
 
-# install pipenv
-RUN pip3 install pipenv
-
-# install python dependencies
+# Install pipenv and Python dependencies with cache cleanup
+RUN pip3 install --no-cache-dir pipenv
 COPY Pipfile* ./
-RUN pipenv sync --system
+RUN pipenv sync --system && pip cache purge
 
-RUN chown -R spark_user:spark /opt/bitnami
+# RUN chown -R spark_user:spark /opt/bitnami
```

Comment:
It creates a 3.6GB layer. I think this was previously required when the image was used as both Spark master and worker. It may no longer be necessary, but I've kept it commented out as a fallback in case any issues arise.

Reply:
Could you put in a comment so the next person knows why this was commented out and when it might need to be restored?

Reply:
👍

Reply:
:+1: